### DISCUSSION TITLE
"Can't set block device" after using rauc

### DISCUSSION THREAD

**Discussion by synaption** [rauc/rauc#971] [General]
I have rootfs updates working, but I am trying to get bootloader updates working and it's complaining that there is not enough room.  Any idea what variable I should change?  I'm using buildroot with an imx8.

```
$ rauc install /tmp/update.raucb 
installing
  0% Installing
  0% Determining slot states
 20% Determining slot states done.
 20% Checking bundle
 20% Verifying signature
 40% Verifying signatu[  211.064279] loop0: detected capacity change from 0 to 126173
re done.
 40% Checking bundle done.
[  211.096389] loop0: detected capacity change from 126173 to 126168
 40% Checking manifest contents
 60% Checking manifest contents done.
[  211.138425] device-mapper: ioctl: 4.45.0-ioctl (2021-03-22) initialised: dm-devel@redhat.com
[  211.155555] device-mapper: verity: sha256 using implementation "sha256-caam"
 60% Determining target install group
 80% Determining target install group done.
 80% Updating slots
 80% Checking slot bootloader.0
 85% Checking slot bootloader.0 done.
 85% Copying image to bootloader.0
 90% Copying image to bootloader.0 failed.
100% Updating slots failed.
100% Installing failed.
LastError: Installation error: Failed updating slot bootloader.0: Size of image (67108864) does not fit to slot size 33554432
idle
Installing `/tmp/update.raucb` failed
```

**Comment by ejoerns** [comment]
It reports that the bootloader does not fit in your currently configured slot.
Please share your bootloader.0 slot configuration. Also make sure that the actual space reserved for the bootloader is twice the size of the maximum bootloader image in case of using MBR/GPT switch.

I assume you have followed the instructions in https://rauc.readthedocs.io/en/latest/advanced.html#updating-the-bootloader ?

**Reply by synaption** [reply]
It turns out that I only needed to make the size of my bootloader to 32MB.  I though there was an error because the noted size limit of 32MB happened to be exactly half of the 64MB I was trying to use.  This turned out to just be a coincidence and there was no actual relationship between the two numbers.
